src/features/TakeNow/ui/TakeNowSuccessModal.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from 'react';
+
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { MuiModal } from '~/shared/ui';
+import { useCustomTranslation, useOnceEffect } from '~/shared/utils';
+
+export const TakeNowSuccessModal = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { t } = useCustomTranslation();
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  useOnceEffect(() => {
+    if (location.state && location.state.showTakeNowSuccessModal && !isOpen) {
+      navigate(window.location.pathname, {
+        state: {
+          ...location.state,
+          showTakeNowSuccessModal: undefined,
+        } as unknown,
```

Comment:
It's because of the spread of `location.state`, which is of type `any`. Similar to this

https://github.com/ChildMindInstitute/mindlogger-web-refactor/blob/cd105a0691f183b715e2426a857e900dea89edf0/src/pages/Login/index.tsx#L35-L41

I'm not sure why asserting the whole object as `unknown` fixes it though 🤷